src/main/java/net/neoforged/neoforge/capabilities/color/ColorApplicationResult.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.capabilities.color;
+
+/**
+ * For usage with {@link IColorable}; represents the result of trying to apply color to something.
+ */
+public enum ColorApplicationResult {
+    /**
+     * Implies that the color was read by the consuming object, and determined that
+     * application will not change state.
+     * Ex: Tried to use black dye on an already-black sheep
+     */
+    ALREADY_APPLIED,
```

Comment:
I'm not sure how this would be different from `Cannot Apply` and could be replaced by `getColor()` in both the existing interfaces if they actually need to respond to the why.